sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -164,7 +164,7 @@ class AdaptiveQueryExecSuite
       val bhj = findTopLevelBroadcastHashJoin(adaptivePlan)
       assert(bhj.size == 3)
       // additional shuffle exchange introduced, only one shuffle reader to local shuffle reader.
-      checkNumLocalShuffleReaders(adaptivePlan, 1)
+      checkNumLocalShuffleReaders(adaptivePlan, 2)
```

Comment:
Just to confirm, would the change make this value consistently be 2? Because the value has changed to 2 but the value was actually flaky depending on the situation/randomness (maybe).

You may want to run the same for what I've discovered, 1) solely in local dev, 2) test suite in local dev, 3) trigger CI for 5 times or alike.

Reply:
@HeartSaVioR  With this patch, the value will consistently be 2. Because we already optimize all the possible local shuffle reader.  And I have run in local dev and also the test suite, the value are all 2. Thanks.

Reply:
OK thanks for confirming.